adapters/ix/ix.go

```diff
@@ -222,6 +224,37 @@ func (a *IxAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalReque
 	return bidderResponse, errs
 }
 
+func getMediaTypeForBid(bid openrtb2.Bid, impMediaTypeReq map[string]openrtb_ext.BidType) (openrtb_ext.BidType, error) {
+	switch bid.MType {
+	case openrtb2.MarkupBanner:
+		return openrtb_ext.BidTypeBanner, nil
+	case openrtb2.MarkupVideo:
+		return openrtb_ext.BidTypeVideo, nil
+	case openrtb2.MarkupAudio:
+		return openrtb_ext.BidTypeAudio, nil
+	case openrtb2.MarkupNative:
+		return openrtb_ext.BidTypeNative, nil
+	}
```

Comment:
Nice use of the new OpenRTB 2.6 `MType` field. I would like to build support directly in PBS-Core for MType. What behavior would you like ideally to see from how PBS-Core handles this field? Maybe a new interface callback func to get the media type if MType doesn't exist? Maybe a bid error where it's rejected if MType doesn't exist and BidType isn't provided? I'm looking for suggestions.

Reply:
> Maybe a bid error where it's rejected if MType doesn't exist  

Not sure if we should do this, because mtype is not a required field in openrtb 2.6 and I believe, most of the exchanges don't support it yet.    
IMO, PBS-Core can implement similar logic as I made here - take mtype if present, else fallback to ext.prebid.type; otherwise bid request's type as last resort (for non-multiformat). In that case, Adapter don't need to handle type anymore.